way_to_home/notification/views.py

```diff
@@ -1 +1,127 @@
-"""This module that provides base logic for CRUD of notification`s model objects."""
+"""
+Notification view module
+================
+
+This module that provides base logic for CRUD of notification`s model objects.
+"""
+
+from django.views import View
+from django.http import HttpResponse, JsonResponse
+
+from notification.models import Notification
+from way.models import Way
+
+
+class NotificationView(View):
+    """
+    Notification view that handles GET, POST, PUT, DELETE requests and provides
+    appropriate operations with notification model.
+    """
+    def get(self, request, way_id, notification_id=None):
+        """ Method that handles GET request. """
+        user = request.user
+        way = Way.get_by_id(way_id)
+
+        if not way:
+            return HttpResponse('way is not found', status=400)
+
+        if way.user != user:
+            return HttpResponse('permission denied', status=403)
+
+        if not notification_id:
+            data = [notification.to_dict() for notification in way.notifications.all()]
+            return JsonResponse(data, status=200, safe=False)
+
+        notification = Notification.get_by_id(notification_id)
+        if not notification:
+            return HttpResponse('Notification not found', status=400)
+
+        if notification.way != way:
+            return HttpResponse('permission denied', status=403)
```

Comment:
If u change step of retrieving notification to:
`notification = way.notifications.filter(id=notification_id)`
You don't need to check this:
`if notification.way != way:`